rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -1483,49 +1483,43 @@ data class CatStmt(
     val left: Expression?,
     val right: Expression,
     val target: AssignableExpression,
-    val blanksInBetween: Int,
+    val blanksInBetween: Expression?,
```

Comment:
It is not clear to me why the change of blanksInBetween from Int to expression, which however is nothing more that a computed number, has required so much changes in this function implementation.  

Have you try to replace the first statement from:
 `val blanksInBetween = blanksInBetween ` to
```
val blanksInBetween = blanksInBetween?.let { 
            interpreter.eval(it) 
        }?.asInt()?.value?.toInt() ?: 0 
```

Reply:
Because `blanksInBetween` is an Expression, not a value and in `parseTreeToAst` phase I don't know the value of the expression because the value may be change at runtime. In the previous version `CAT` statement don't accept a variable after `:` in factor 2.